src/zkevm_specs/opcode/mul.py

```diff
@@ -0,0 +1,56 @@
+from typing import Sequence
+
+from ..encoding import U8, is_circuit_code, U256, u8s_to_u64s
+
+
+def mul_common(
+    a8s: Sequence[U8],
+    x8s: Sequence[U8],
+    y8s: Sequence[U8],
+    t0: U256,
+    t1: U256,
+    t2: U256,
+    t3: U256,
+    v0: Sequence[U8],
+    v1: Sequence[U8],
+):
+    assert len(a8s) == len(x8s) == len(y8s) == 32
+    assert len(v0) == len(v1) == 9
```

Comment:
`specs/opcode/02MUL.md` specifies that v0 and v1 are between 0 and 2^64.  But I notice that here you assert that v0 and v1 are 9 elements of 8 bits, which means between 0 and 2^72.  Am I missing something?

Reply:
t1 is the sum of two 128bit products from limbs so they would excceed 128bit, which lead to the possibility that v0 would excceed 64bit. So does v1 and in fact it should not exceed 2^66 instead of 2^64. We can relax the constraint to make v0/v1 not exceed 2^72 or 9 bytes because everything should be ok only if v0 (or v1) * 2^128 not exceed the prime field we used (for bn254 it is about 253bit).

I should update the narration from its reference in the following patch.